internal/object/object.go

```diff
@@ -120,11 +124,28 @@ func FromEvent(ctx context.Context, db *database.DB, ev *event.Event) (*Object,
 	cacheMu.Lock()
 	defer cacheMu.Unlock()
 
-	object, ok := cache[id.String()]
-	if !ok {
-		object = New(db, ev)
-		object.ID = id
-		cache[id.String()] = object
+	newObject := new(Object)
+	object, objectExists := cache[id.String()]
+	if !objectExists {
+		newObject = New(db, ev)
+		newObject.ID = id
+		if !newObject.Muted.Valid {
+			newObject.Muted = types.Bool{Bool: false, Valid: true}
+		}
+	} else {
+		*newObject = *object
+
+		newObject.ExtraTags = ev.ExtraTags
+		newObject.Name = ev.Name
+		newObject.URL = utils.ToDBString(ev.URL)
+
+		if ev.Mute.Valid && (!ev.Mute.Bool || !newObject.IsMuted()) {
+			// This behaviour is explicitly requested so that two successive events with `mute` set
+			// to true don't influence each other, even if their `mute_reason` differs. We should
+			// only care about the first mute event.
+			newObject.Muted = ev.Mute
+			newObject.MuteReason = utils.ToDBString(ev.MuteReason)
+		}
```

Comment:
> This behaviour is explicitly requested

What would you do here if it wasn't?

Maybe another misunderstanding/miscommunication here. I didn't want to say "`mute_reason` may not be updated under any circumstances" but rather something like "I don't think we want to put the effort into providing useful events/notifications like 'the mute reason changed from being in downtime to being in downtime and acknowledged' as that would require keeping more detailed track of the mute reasons and sending a notification like 'the muting reason changed from downtime to acknowledgement' would be pointless if the downtime still continues to exist".